src/middlewared/middlewared/alert/source/rsync.py

```diff
@@ -35,3 +35,40 @@ async def delete(self, alerts, query):
             lambda alert: alert.key != str(query),
             alerts
         ))
+
+
+class RsyncTaskLockedAlertClass(AlertClass, OneShotAlertClass):
+    deleted_automatically = False
+
+    category = AlertCategory.TASKS
+    level = AlertLevel.WARNING
+    title = 'Rsync Task Locked'
+    text = 'Rsync task operating on "%(path)s" path is using a locked resource. Please disable the task.'
+
+    async def create(self, args):
+        return Alert(RsyncTaskLockedAlertClass, args, key=args['id'])
+
+    async def delete(self, alerts, query):
+        return list(filter(
+            lambda alert: alert.key != str(query),
+            alerts
+        ))
+
+
+class RsyncModuleLockedAlertClass(AlertClass, OneShotAlertClass):
+    deleted_automatically = False
+
+    category = AlertCategory.SHARING
+    level = AlertLevel.WARNING
+    title = 'Rsync Module Locked'
+    text = 'Rsync module "%(name)s"  operating on "%(path)s" path is using a locked ' \
+           'resource. Please disable the module.'
```

Comment:
Why are we operating on all of these? We should not operate at all and ask the user to unlock the resource, not the other way around.